src/doc/examples/lock-nodes.md

```diff
@@ -0,0 +1,184 @@
+# Examples
+
+## Node depended resources
+
+Create resources that are node depended is in many cases very helpful. That means a job must pick a target node where the requested resource is available.
+
+```groovy
+// allocate node
+node('some-build-node') {
+  // Lock resource named *whatever-resource-some-build-node*
+  lock("whatever-resource-${env.NODE_NAME}") {
+    echo "I am no node ${env.NODE_NAME} and locked resource ${env.LOCKED_RESOURCE}"
+  }
+}
+```
+
+But much more useful is lock node first.
+
+```groovy
+// Lock resource named *some-build-node*
+lock('some-build-node') {
+  // allocate node
+  node(env.LOCKED_RESOURCE) {
+
+    echo "I am no node ${env.NODE_NAME} and locked resource ${env.LOCKED_RESOURCE}"
+  }
+}
+```
+
+Let explain in more complex use case.
+
+*Request:*
+Our job tests server-client integration. That means we need 2 nodes (1 server and 1 client).
+On every node must be test sources up-to-date.
+Tests are running only on client side.
+
+*Solution:*
+Create 2 nodes:
+
+- node-server
+- node-client
+
+and execute it parallel like this:
+
+```groovy
+Map stages = [:];
+stages['server'] = {
+  node('node-server') {
+    prepareTests()
+    startServer()
+  }
+}
+stages['client'] = {
+  node('node-client') {
+    prepareTests()
+    startClientTest()
+  }
+}
+// execute all prepare stages synchronous
+parallel stages
+
+// Prepare tests on node
+void prepareTests() {
+  checkout([$class: 'GitSCM',
+            branches: [[name: 'master']]
+           ])
```

Comment:
I think that we don't want to encode the SCM type or the branch name here because that is provided by the multibranch project where we are running this job.  By using a multibranch job, that allows me to test changes to the environment without altering the source code of the test.  It will use the branch that provided the Jenkinsfile
```suggestion
  checkout scm
```